rampwf/score_types/detection.py

```diff
@@ -239,55 +214,99 @@ def scp_single(y_true, y_pred, shape, minipatch=None):
     return score, n_true, n_pred
 
 
-def ospa_single(y_true, y_pred, cut_off=1, minipatch=None):
+def ospa(y_true, y_pred, minipatch=None):
+    """
+    Optimal Subpattern Assignment (OSPA) metric for IoU score.
+
+    This metric provides a coherent way to compute the miss-distance
+    between the detection and alignment of objects. Among all
+    combinations of true/predicted pairs, if finds the best alignment
+    to minimise the distance, and still takes into account missing
+    or in-excess predicted values through a cardinality score.
+
+    The lower the value the smaller the distance.
+
+    Parameters
+    ----------
+    y_true, y_pred : list of list of tuples
+    minipatch : list of int, optional
+        Bounds of the internal scoring patch (default is None)
+
+    Returns
+    -------
+    float: distance between input arrays
+
+    References
+    ----------
+    http://www.dominic.schuhmacher.name/papers/ospa.pdf
+
+    """
+    ospas = np.array([ospa_single(t, p, minipatch)
+                      for t, p in zip(y_true, y_pred)])
+
+    ious, n_count, n_total = ospas.sum(axis=0)
+
+    if n_total == 0:
+        return 0
+
+    return ious * n_count / n_total
+
+
+def ospa_single(y_true, y_pred, minipatch=None):
     """
     OSPA score on single patch. See docstring of `ospa` for more info.
 
     Parameters
     ----------
     y_true, y_pred : ndarray of shape (3, x)
         arrays of (x, y, radius)
-    cut_off : float, optional (default is 1)
-        penalizing value for wrong cardinality
     minipatch : list of int, optional
         Bounds of the internal scoring patch (default is None)
 
     Returns
     -------
-    float: distance between input arrays
+    (iou_sum, n_pred, n_total)
+        float - sum of ious of matched entries
+        int - sum of matched entries
```

Comment:
sum -> number ? (and below as well)

Reply:
I don't get your comment

Reply:
Meaning of my cryptic comment: "I *think* 'sum' should be replaced by 'number' in that line"

Just because a "sum of entries" is a bit a strange thing to calculate, you cannot really sum them. But you can count them. Therefore I suggested "number".